src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java

```diff
@@ -182,6 +205,112 @@ public void checkTrialAccounts() {
+                    String description = googleDoc.select(".kCrYT > div > div > div > div > div").text();
+                    if (description.indexOf(q) == -1)
+                        googleCount = 0;
+                }
+            }
+            catch (Exception e){
+                e.printStackTrace();
+            }
+            
+            int baiduCount = 0;
+            try{
+                Document document = Jsoup.connect(String.format("https://www.baidu.com/s?wd=\"%s\"", q)).get();
+                Elements elements = document.select("div.result.c-container");
+                baiduCount = elements.size();
+            }
+            catch (Exception e){
+                e.printStackTrace();
+            }          
+
+            if (githubCount > 0 || googleCount > 0 || baiduCount > 0){   
+                ExposedToken t = new ExposedToken();
+                UserDTO user = userMapper.userToUserDTO(token.getUser());
+                t.setToken(q);
+                t.setEmail(user.getEmail());
+                t.setFirstName(user.getFirstName());
+                t.setLastName(user.getLastName());
+                t.setLicenseType(user.getLicenseType() != null ? user.getLicenseType().name() : "");
+                List<String> source = new ArrayList<>();
+                if(githubCount > 0){
+                    source.add("GitHub");
+                }
+                if (googleCount > 0){
+                    source.add("Google");
+                }
+                if (baiduCount > 0){
+                    source.add("Baidu");
+                }
+                t.setSource(source.stream().collect(Collectors.joining(", ")));   
+                results.add(t);
+
+                updateExposedToken(token);
+                mailService.sendExposedTokensInfoMailToUser(user, t);
+            }
+            sleep(1000);
+        }
+        log.info("Searching exposed tokens pipeline finished!");
+        if (results.size() > 0){
+            mailService.sendExposedTokensInfoMail(results);
+        }
+    }
+
+    private HttpResponse getGoogleResponse(String query) {
+        String uri = String.format("https://www.google.com/search?q=%s", query);
+        HttpClient client = HttpClientBuilder.create().build();
+        HttpGet request = new HttpGet(uri);
+        HttpResponse result = null;
+        try {
+          result = client.execute(request);
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+        return result;
+    }
+
+    private void updateExposedToken(Token token){
+        Token newToken = tokenProvider.createToken(token.getUser(), Optional.of(token.getExpiration()), Optional.of(token.isRenewable()));
+        newToken.setCreation(token.getCreation());
+        newToken.setCurrentUsage(token.getCurrentUsage());
+        newToken.setUsageLimit(token.getUsageLimit());
+        token.setExpiration(Instant.now());
+        tokenService.save(token);
+        tokenService.save(newToken);
```

Comment:
Let's create a new createToken  method in the tokenProvider to take a Token as a input to avoid the multiple save.